### DISCUSSION TITLE
ESP32-WROOM-E with Ethernet Firmware

### DISCUSSION THREAD

**Discussion by jingqunhu** [nanoframework/Home#1398] [Q&A]
I need to use the ESP32-WROOM-E module with Ethernet, but after brushing the firmware of lilygo or OLIMEX, the device does not work and keeps rebooting. I would like to ask that lilygo and OLIMEX are better suited to the esp32 with Ethernet module. Thank you so much for your support

**Comment by jingqunhu** [comment]
When I use ESP32-WROOM-E to brush into ESP32-OLIMEX, the module repeated restart print is as follows:
rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:5632
load:0x40078000,len:14644
ho 0 tail 12 room 4
load:0x40080400,len:3708
entry 0x4008067c
I (29) boot: .NET nanoFramework 2nd stage bootloader ESP-IDF 3572900
I (29) boot: build Nov  4 2023 00:19:19
I (30) boot: chip revision: v3.1
I (34) boot.esp32: SPI Speed      : 80MHz
I (39) boot.esp32: SPI Mode       : DIO
I (43) boot.esp32: SPI Flash Size : 16MB
I (48) boot: Enabling RNG early entropy source...
I (53) boot: Partition Table:
I (57) boot: ## Label            Usage          Type ST Offset   Length
I (64) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (72) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (79) boot:  2 factory          factory app      00 00 00010000 001a0000
I (87) boot:  3 deploy           Unknown data     01 84 001b0000 002e0000
I (94) boot:  4 config           Unknown data     01 82 00490000 00300000
I (102) boot: End of partition table
I (106) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=2c018h (180248) map
I (168) esp_image: segment 1: paddr=0003c040 vaddr=3ff80067 size=00008h (     8) load
I (169) esp_image: segment 2: paddr=0003c050 vaddr=3ffb0000 size=034c4h ( 13508) load
I (179) esp_image: segment 3: paddr=0003f51c vaddr=40080000 size=00afch (  2812) load
I (184) esp_image: segment 4: paddr=00040020 vaddr=400d0020 size=d0774h (853876) map
I (446) esp_image: segment 5: paddr=0011079c vaddr=40080afc size=1b584h (112004) load
I (485) esp_image: segment 6: paddr=0012bd28 vaddr=400c0000 size=00068h (   104) load
I (499) boot: Loaded app from partition at offset 0x10000
I (499) boot: Disabling RNG early entropy source...
I (511) psram: This chip is ESP32-D0WD
I (511) spiram: Found 64MBit SPI RAM device
I (511) spiram: SPI RAM mode: flash 80m sram 40m
I (516) spiram: PSRAM initialized, cache is in low/high (2-core) mode.
I (523) cpu_start: Pro cpu up.
I (527) cpu_start: Starting app cpu, entry point is 0x40086030
I (519) cpu_start: App cpu up.
I (549) cpu_start: Pro cpu start user code
I (549) cpu_start: cpu freq: 240000000
I (549) cpu_start: Application information:
I (554) cpu_start: Project name:     nanoCLR
I (558) cpu_start: App version:      1.8.1.671
I (564) cpu_start: Compile time:     Oct 22 2023 00:15:44
I (570) cpu_start: ELF file SHA256:  3f15d101f8e81af6...
I (576) cpu_start: ESP-IDF:          ac5d805
I (581) cpu_start: Min chip rev:     v0.0
I (585) cpu_start: Max chip rev:     v3.99 
I (590) cpu_start: Chip rev:         v3.1
I (595) heap_init: Initializing. RAM available for dynamic allocation:
I (602) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (608) heap_init: At 3FFBC8A8 len 00023758 (141 KiB): DRAM
I (614) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (621) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (627) heap_init: At 4009C080 len 00003F80 (15 KiB): IRAM
I (633) spiram: Adding pool of 4095K of external SPI memory to heap allocator
I (642) spi_flash: detected chip: gd
I (645) spi_flash: flash io: dio
I (661) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (661) spiram: Reserving pool of 32K of internal memory for DMA/internal allocations
Ɵ蟷د­&(FCnZBɎC顂ONBj烂ՂBJՎKբBJNBJNbɢBJ^BjN៊ҝJvBNBBՊJc񛱚Zɚ陊ZZ񐅧_K罥ts Jul 29 2019 12:21:46

**Reply by josesimoes** [reply]
Please switch baud rate to 921600 to see the messages after the CLR boots.

**Reply by jingqunhu** [reply]
ELF file SHA256: c89c4f8ceb92ddd7

Rebooting...
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x4019ac69  PS      : 0x00060e30  A0      : 0x800d6b44  A1      : 0x3ffcdda0  
A2      : 0x3f80087c  A3      : 0x00000001  A4      : 0xaaaaaaab  A5      : 0xffffffff  
A6      : 0x3f800884  A7      : 0xffffffff  A8      : 0xdf010000  A9      : 0xffffffff  
A10     : 0x00000001  A11     : 0x80000002  A12     : 0x800934c0  A13     : 0x3ffcdca0  
A14     : 0x00000003  A15     : 0x00060023  SAR     : 0x00000015  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000003  LBEG    : 0x4008ca52  LEND    : 0x4008ca5d  LCOUNT  : 0x00000000  


Backtrace: 0x4019ac66:0x3ffcdda0 0x400d6b41:0x3ffcddc0 0x400d6bb2:0x3ffcdde0 0x400d6c3c:0x3ffcde00 0x400d6c86:0x3ffcde20 0x400e38b8:0x3ffcde40 0x400faf05:0x3ffcde70 0x400f956a:0x3ffcde90

**Reply by jingqunhu** [reply]
rt: App version:      1.8.1.713
I (570) cpu_start: Compile time:     Nov  4 2023 00:18:49
I (576) cpu_start: ELF file SHA256:  c89c4f8ceb92ddd7...
I (582) cpu_start: ESP-IDF:          3572900
I (587) cpu_start: Min chip rev:     v0.0
I (591) cpu_start: Max chip rev:     v3.99 
I (596) cpu_start: Chip rev:         v3.1
I (601) heap_init: Initializing. RAM available for dynamic allocation:
I (608) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (614) heap_init: At 3FFBC938 len 000236C8 (141 KiB): DRAM
I (620) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (627) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (633) heap_init: At 4009C71C len 000038E4 (14 KiB): IRAM
I (640) spiram: Adding pool of 4095K of external SPI memory to heap allocator
I (648) spi_flash: detected chip: gd
I (652) spi_flash:@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x4019ac69  PS      : 0x00060e30  A0      : 0x800d6b44  A1      : 0x3ffcdda0  
A2      : 0x3f80087c  A3      : 0x00000001  A4      : 0xaaaaaaab  A5      : 0xffffffff  
A6      : 0x3f800884  A7      : 0xffffffff  A8      : 0xdf090000  A9      : 0xffffffff  
A10     : 0x00000001  A11     : 0x80000002  A12     : 0x800934c0  A13     : 0x3ffcdca0  
A14     : 0x00000003  A15     : 0x00060023  SAR     : 0x00000015  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000003  LBEG    : 0x4008ca52  LEND    : 0x4008ca5d  LCOUNT  : 0x00000000  


Backtrace: 0x4019ac66:0x3ffcdda0 0x400d6b41:0x3ffcddc0 0x400d6bb2:0x3ffcdde0 0x400d6c3c:0x3ffcde00 0x400d6c86:0x3ffcde20 0x400e38b8:0x3ffcde40 0x400faf05:0x3ffcde70 0x400f956a:0x3ffcde90

**Reply by josesimoes** [reply]
Is this a local firmware build?

**Reply by jingqunhu** [reply]
no，I'm using firmware from the cloud

**Reply by josesimoes** [reply]
Than it has to be something with the code you're running. Can you deploy a simple hello world application?

**Reply by jingqunhu** [reply]
The device keeps restarting, the serial port cannot communicate, and no device can be connected to the device explorer

**Reply by josesimoes** [reply]
Then you're using the wrong firmare. Check the home page for the correct target name to use with your board.

**Reply by jingqunhu** [reply]
ESP32 (ESP32-D0WD-V3 (revision v3.1))
Features WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Flash size 16MB GD25Q128 from GIGADEVICE (manufacturer 0x200 device 0x16408)
PSRAM: undetermined
Crystal 40MHz
MAC B0:B2:1C:31:C5:A4

**Reply by jingqunhu** [reply]
Firmware with Ethernet, currently only lilygo and OLIMEX

**Reply by jingqunhu** [reply]
@josesimoes

**Reply by josesimoes** [reply]
You may have to do a local build as we don't' offer all possible combinations of the features.
Check [here](https://docs.nanoframework.net/content/building/build-esp32.html)

**Comment by jingqunhu** [comment]
I need to use Ethernet, which ESP32 module is more suitable, because when ota reflection assembly, need enough memory space, so the cpu built-in ram is not enough, need external psram.
